tests/filecheck/transforms/accfg-insert-resets.mlir

```diff
@@ -0,0 +1,111 @@
+// RUN: snax-opt --split-input-file -p accfg-insert-resets %s | filecheck %s
+
+func.func @args(%state: !accfg.state<"simple">) {
+    return
+}
+
+// CHECK-LABEL: @args
+// CHECK-SAME:  ([[state:%\S+]] : !accfg.state<"simple">)
+// CHECK:       accfg.reset [[state]]
+
+
+// -----
+
+func.func @returned_args(%state: !accfg.state<"simple">) -> !accfg.state<"simple"> {
+    return %state : !accfg.state<"simple">
+}
+
+// CHECK-LABEL: @returned_args
+// CHECK-SAME:  ([[state:%\S+]] : !accfg.state<"simple">)
+// CHECK-NOT:       accfg.reset [[state]]
+
+
+// -----
+
+func.func @simple(%i : i32) {
+    %state = accfg.setup "acc" to ("i" = %i : i32) : !accfg.state<"acc">
+    return
+}
+
+// CHECK-LABEL: @simple
+// CHECK-NEXT: [[state:%\S+]] = accfg.setup "acc" to ("i" = %i : i32) : !accfg.state<"acc">
+// CHECK:      accfg.reset [[state]] : !accfg.state<"acc">
+
+
+// -----
+
+func.func @with_uses(%i : i32) {
+    %state = accfg.setup "acc" to ("i" = %i : i32) : !accfg.state<"acc">
+
+    %t = "accfg.launch"(%state) <{"param_names" = [], "accelerator" = "acc"}> : (!accfg.state<"acc">) -> !accfg.token<"acc">
+    "accfg.await"(%t) : (!accfg.token<"acc">) -> ()
+    return
+}
+
+// CHECK-LABEL: @with_uses
+// CHECK-NEXT: [[state:%\S+]] = accfg.setup "acc" to ("i" = %i : i32) : !accfg.state<"acc">
+// CHECK-NEXT: [[token:%\S+]] = "accfg.launch"(%state)
+// CHECK-NEXT: accfg.reset [[state]] : !accfg.state<"acc">
+// CHECK-NEXT: "accfg.await"([[token]])
```

Comment:
The `CHECK` lines don't need to match the whole line

Reply:
I mainly thought it was weird that you already reset the accelerator's registers before the await.
I think we should be careful with this, since there could (and should) be hardware support for clearing all the registers (also the shadow registers) upon reset. That would make this code invalid if compiled. We should be very clear on semantics here, is all that I'm saying.
it is valid if the shadow registers are kept, but might lead to weird results if the reset affects them or other accelerator parameters. Maybe `reset` is too broad as a term?

Reply:
Added a flag `reset-after-await` to pass, will need some more fine tuning, but already most of the way there I think